lib/honeybadger/agent/worker.rb

```diff
@@ -38,7 +38,7 @@ def initialize(config, feature)
         @throttles = []
         @mutex = Mutex.new
         @marker = ConditionVariable.new
-        @queue = Queue.new(1000)
+        @queue = Queue.new(config.max_queue_size)
```

Comment:
The default for this value should be added to [lib/honeybadger/config/defaults.rb](https://github.com/honeybadger-io/honeybadger-ruby/blob/master/lib/honeybadger/config/defaults.rb#L73), then you can just do `@queue = Queue.new(config[:max_queue_size])` without the Config#max_queue_size helper method.